particles/fcc/particle.py

```diff
@@ -45,3 +46,10 @@ def short_info(self):
             e = self.e()        
         )
 
+    def  dagid_str(self):
```

Comment:
same, shouldn't this be in BaseParticle? 
that would avoid code duplication

Reply:
I would even think that this should be in the Identifier module. 
There could be an identifier class, with a __repr__ method

Reply:
I have created a static id_str member in Identifier and called this from pfobjects particle, and from tlv particle. This minimises any duplication.
I didn't create the repr because the Identifier is a long, and the class has only static members.
I didn't move into the base particle because there is no requirement for dagid to be a long papas Identifier at this level.